File: frontend/src/types/shopping.ts
export interface ShoppingListItem {
  name: string;
  quantity: number;
  notes?: string;
}

export interface ShoppingList {
  id: number;
  name: string;
  items: Array<{ name: string; quantity: number }>;
}
export interface CreateShoppingListData {
  name: string;
  items: ShoppingListItem[];
}

export interface UpdateShoppingListData {
  name?: string;
  items?: ShoppingListItem[];
  is_active?: boolean;
}

File: services/api_gateway/app/schemas/shopping.py
from typing import List, Optional
from pydantic import BaseModel

class ShoppingListItem(BaseModel):
    name: str
    quantity: int
    notes: Optional[str] = None

class ShoppingList(BaseModel):
    id: int
    name: str
    items: List[ShoppingListItem]

class ShoppingListCreate(BaseModel):
    name: str
    items: List[ShoppingListItem]